analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ImplementSerializationMethodsCorrectly.CSharp8.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+public interface IWithValidSerializationMethods
+{
+    [OnSerializing]
+    protected abstract void OnSerializingMethod(StreamingContext context); // Compliant
+
+    [OnSerialized]
+    protected abstract void OnSerializedMethod(StreamingContext context); // Compliant
+
+    [OnDeserializing]
+    protected void OnDeserializingMethod(StreamingContext context) { } // Compliant
+
+    [OnDeserialized]
+    protected virtual void OnDeserializedMethod(StreamingContext context) { } // Compliant
+}
+
+public interface IWithInvalidSerializationMethodsModifiers
+{
+    [OnSerializing]
+    public abstract void OnSerializingMethod(StreamingContext context); // Compliant
```

Comment:
Also kept on the first serialization method of `IWithInvalidSerializationMethodsModifiers`, since it is compliant for the same reason, but it was non-compliant in the past for a different one (namely, because of custom method parameters).